app-backend/db/src/main/scala/AnnotationDao.scala

```diff
@@ -116,20 +160,42 @@ object AnnotationDao extends Dao[Annotation] {
         geometry = ${annotation.geometry},
         annotation_group = ${annotation.annotationGroup},
         labeled_by = ${annotation.labeledBy},
-        verified_by = ${annotation.verifiedBy}
+        verified_by = ${annotation.verifiedBy},
+        project_layer_id = ${annotation.projectLayerId}
       WHERE
-        id = ${annotation.id}
+        id = ${annotation.id} AND project_id = ${projectId}
     """).update.run
   }
 
-  def listProjectLabels(projectId: UUID): ConnectionIO[List[String]] = {
-    (fr"SELECT DISTINCT ON (label) label FROM" ++ tableF ++ Fragments
-      .whereAndOpt(
-        Some(fr"project_id = ${projectId}")
-      )).query[String].to[List]
+  def listProjectLabels(
+      projectId: UUID,
+      projectLayerIdO: Option[UUID] = None): ConnectionIO[List[String]] = {
+    for {
```

Comment:
Since we're adding an `exportAll` query parameter elsewhere, it might make sense to add something similar to the other project endpoints. I don't feel super strongly about this though, so I'd be fine with just creating an issue to discuss / punt on it until there's an actual use-case / user request for it.

Reply:
An issue is created here: https://github.com/raster-foundry/raster-foundry/issues/4563